### DISCUSSION TITLE
How to pass `PROTOC` environment variable to `tonic-health`?

### DISCUSSION THREAD

**Discussion by tamuhey** [google/cargo-raze#443] [Q&A]
Sample repository: https://github.com/tamuhey/bazel_tonic_sample

I want to use [tonic-health](https://crates.io/crates/tonic-health) crate, which uses `protoc` in its `build.rs`.
Now, `bazel build` is failing with the following error:

```
Error: Custom { kind: NotFound, error: "failed to invoke protoc (hint: https://docs.rs/prost-build/#sourcing-protoc): No such file or directory (os error 2)" }
```

This indicates `build.rs` couldn't find `protoc`.
I can pass the build if I manually edit `BUILD.tonic-health.bazel` which is generated by `cargo-raze` and add `PROTOC` env var and `protoc` data into `:tonic_health_build_script`, but I don't want to edit the machine-generated file.
And I know I can pass the `protoc` binary to the build environment with [CrateSettings.data_dependencies](https://github.com/google/cargo-raze/blob/1ff0a79e89101e720e5a2c6c9f56a0d73367a1cf/impl/src/settings.rs#L180).
However, I don't know how to set the binary path to `PROTOC` env var.
How to do that?

**Comment by GregBowyer** [comment]
This is not a good answer.

I am working on getting tonic to work with rules_proto_grpc as fairly out of the box. I might be messing with this and have a better answer when I am further down that road.

**Comment by redshirtrob** [comment]
Here's how I made this work:

```
[package.metadata.raze.crates.tonic-health.'*']
build_tools_dependencies = [
    "@//:protoc",
]

[package.metadata.raze.crates.tonic-health.'*'.buildrs_additional_environment_variables]
PROTOC = "$(location @//:protoc)"
```
